core/genesis.go

```diff
@@ -341,15 +341,16 @@ func SetupGenesisBlockWithOverride(db ethdb.Database, triedb *trie.Database, gen
 		if hash != stored {
 			return genesis.Config, hash, &GenesisMismatchError{stored, hash}
 		}
+		applyOverrides(genesis.Config)
```

Comment:
> just to confirm, this apply operation is moved before the genesis.Commit so that the overridden chain config will be persisted into database, right?

it's one reason, the other one is that verkle mode is determined by `pragueTime`/`verkleTime`. If an override has been set, but the override hasn't been activated when calling `Commit` then `flush` will create a MPT instead of a verkle tree.

> should we also do override first and then commit?

yes, good catch.

Reply:
yes, good catch